### PR TITLE
Feature/pwn 7616 update send flow

### PR DESCRIPTION
**Pull request by Davran331122**
1. ## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7616

## Description of Work

[ what work was done in this ticket ]

## Screenshots (Optional)

| Screenshot #1 | Screenshot #2 | Screenshot #3 |
|:-------------:|:-------------:|:-------------:|
| [Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3] |

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt [hunk 1]
```diff
@@ -217,7 +217,7 @@ class NewSendPresenter(
     private suspend fun initializeFeeRelayer(
         view: NewSendContract.View,
         initialToken: Token.Active,
-        solToken: Token.Active
+        solToken: Token.Active,
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt:220]: wtf?

File: app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt [hunk 2]
```diff
@@ -195,7 +195,7 @@ class NewSendPresenter(
 
     private fun handleFeeRelayerStateUpdate(
         newState: FeeRelayerState,
-        view: NewSendContract.View
+        view: NewSendContract.View,
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt:198]: wtf?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -29,7 +29,7 @@
         android:fullBackupContent="@xml/backup_rules"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
-        android:largeHeap="false"
+        android:largeHeap="true"
```
Comment (gslevinkov) [inline app/src/main/AndroidManifest.xml:32]: revert and do something with emulator please

File: app/src/main/java/org/p2p/wallet/bridge/send/BridgeSendModule.kt [hunk 4]
```diff
@@ -17,5 +24,19 @@ object BridgeSendModule : InjectionModule {
...
         factoryOf(::EthereumSendRemoteRepository) bind EthereumSendRepository::class
         factoryOf(::BridgeSendInteractor)
+        factoryOf(::SendUiMapper)
+        factory {
+            val handlers = setOf(
+                AmountChangeActionHandler(),
+                InitFeatureActionHandler(),
+                NewTokenActionHandler(),
+                RefreshFeeActionHandler(get()),
+                SetupTokenHandler(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/BridgeSendModule.kt:34]: can be created like
new(:: SetupTokenHandler)

File: app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendState.kt [hunk 5]
```diff
@@ -1,52 +1,50 @@
...
-        data class Feature(
-            override val lastStaticState: Static,
-            val featureException: SendFeatureException,
-        ) : Exception
+        data class SnackbarMessage(val messageResId: Int) : Exception
+        object FeeLoading : Exception
     }
 
-    sealed interface Static : SendState {
+    sealed interface Event : SendState {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendState.kt:21]: SendEvent

File: app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendState.kt [hunk 6]
```diff
@@ -1,52 +1,50 @@
...
 
 import java.math.BigDecimal
+import org.p2p.core.token.Token
+import org.p2p.wallet.bridge.send.model.BridgeSendFees
 import org.p2p.wallet.bridge.send.statemachine.model.SendFee
-import org.p2p.wallet.bridge.send.statemachine.model.SendToken
-
-sealed interface SendBridgeState
 
 sealed interface SendState {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendState.kt:8]: is it for bridge?
BridgeSendState it is.

File: app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendStateMachine.kt [hunk 7]
```diff
@@ -31,72 +33,51 @@ class SendStateMachine(
...
     fun newAction(action: SendFeatureAction) {
-        refreshFeeTimer?.cancel()
-        lastAction = action
-
-        val staticState = state.lastStaticState
-        val actionHandler = handlers.firstOrNull { it.canHandle(action, staticState) } ?: return
-
-        actions.tryEmit(action to actionHandler)
+        launch {
+            Timber.tag("_______").d("Action = $action")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendStateMachine.kt:61]: revert

File: app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendStateMachine.kt [hunk 8]
```diff
@@ -31,72 +33,51 @@ class SendStateMachine(
...
-            }
-            is Exception -> {
-                val lastStaticState = state.value.lastStaticState
-                val wrappedState = SendState.Exception.Other(
-                    lastStaticState,
-                    SendFeatureException.FeeLoadingError,
-                )
-                flowCollector.emit(wrappedState)
-            }
+            is Exception -> {}
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/statemachine/SendStateMachine.kt:70]: ?

File: app/src/main/java/org/p2p/wallet/bridge/send/statemachine/handler/bridge/RefreshFeeActionHandler.kt [hunk 9]
```diff
@@ -1,20 +1,40 @@
...
         newEvent is SendFeatureAction.RefreshFee
 
     override fun handle(
-        lastStaticState: SendState.Static,
-        newAction: SendFeatureAction
-    ): Flow<SendState> = feeLoader.updateFee(lastStaticState)
+        newAction: SendFeatureAction,
+    ): Flow<SendState> = flow {
+        try {
+            Timber.tag("_____EMIT").d("TRUE")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/statemachine/handler/bridge/RefreshFeeActionHandler.kt:22]: revert?

File: app/src/main/java/org/p2p/wallet/bridge/send/statemachine/handler/bridge/RefreshFeeActionHandler.kt [hunk 10]
```diff
@@ -1,20 +1,40 @@
...
+            val fee = repository.getSendFee(
+                userWallet = action.userWallet,
+                recipient = action.recipient,
+                mint = action.mintAddress,
+                amount = action.amount
+            )
+            Timber.tag("_____EMIT").d("FALSE")
+            emit(SendState.Event.UpdateFee(fee))
+        } catch (e: Throwable) {
+            emit(SendState.Exception.FeeLoading)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/statemachine/handler/bridge/RefreshFeeActionHandler.kt:34]: Timber.i?

File: app/src/main/java/org/p2p/wallet/bridge/send/statemachine/handler/bridge/SetupTokenHandler.kt [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    override fun handle(
+        newAction: SendFeatureAction,
+    ): Flow<SendState> = flow {
+
+        val action = newAction as SendFeatureAction.SetupInitialToken
+        val userTokens = userInteractor.getNonZeroUserTokens()
+            .filter { it.mintAddress in supportedTokensMints }
+            .ifEmpty {
+                // TODO PWN-7613 also block button as we can't send we do not have funds
+                val usdCet = userInteractor.findTokenDataByAddress(ERC20Tokens.USDC.mintAddress) as Token.Other
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/statemachine/handler/bridge/SetupTokenHandler.kt:35]: usdcEt?

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendFragment.kt [hunk 12]
```diff
@@ -70,6 +61,13 @@ class BridgeSendFragment :
     private val initialToken: Token.Active? by args(ARG_INITIAL_TOKEN)
     private val inputAmount: BigDecimal? by args(ARG_INPUT_AMOUNT)
     private val openedFrom: SendOpenedFrom by args(ARG_OPENED_FROM)
+    private val widgetDelegate: UiKitSendDetailsWidgetContract by lazy {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendFragment.kt:64]: by unsafeLazy

**Comment by kabramovp2p** [comment]
I think we can close this PR

**Pull request closed without merging**